hathor/p2p/sync_mempool.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2020 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import deque
+from typing import TYPE_CHECKING, Deque, Generator, List, Optional, Set
+
+from structlog import get_logger
+from twisted.internet.defer import Deferred, inlineCallbacks
+
+from hathor.transaction import BaseTransaction
+from hathor.transaction.storage.exceptions import TransactionDoesNotExist
+
+if TYPE_CHECKING:
+    from hathor.p2p.node_sync_v2 import NodeBlockSync
+
+logger = get_logger()
+
+
+class SyncMempoolManager:
+    """Manage the sync-v2 mempool with one peer.
+    """
+    def __init__(self, sync_manager: 'NodeBlockSync'):
+        """Initialize the sync-v2 mempool manager."""
+        # Shortcuts.
+        self.sync_manager = sync_manager
+        self.manager = self.sync_manager.manager
+        self.tx_storage = self.manager.tx_storage
+        self.reactor = self.sync_manager.reactor
+
+        # Set of tips we know but couldn't add to the DAG yet.
+        self.missing_tips: Set[bytes] = set()
+
+        # Stack used by the DFS in the dependencies.
+        # We use a deque for performance reasons.
+        self.deque: Deque[BaseTransaction] = deque()
+
+        # Maximum number of items in the DFS.
+        self.MAX_STACK_LENGTH: int = 1000
+
+        # Internal variables.
+        # This is used to keep track of the latest requested tx.
+        self._download_tx_deferred: Optional[Deferred] = None
+        self._download_tx_hash: Optional[bytes] = None
+
+    def start(self):
+        pass
```

Comment:
Create stop and protect against unbalanced calls.

Reply:
Or remove start?

Reply:
It was removed.